std/format.d

```diff
@@ -3920,6 +3920,11 @@ if (is(T == class) && !is(T == enum))
             {
                 formatObject(w, val, f);
             }
+            else static if (isForwardRange!T)
+            {
+                auto tmp = val.save;
+                formatRange(w, tmp, f);
```

Comment:
can't you pass `val.save` as parameter directly, i.e w/o tmp ?

Reply:
It's a ref parameter, so no.